File: 0395-longest-substring-with-at-least-k-repeating-characters/0395-longest-substring-with-at-least-k-repeating-characters.cpp
class Solution {
public:
    int longestSubstring(string s, int k) {
        int n = s.size();
        unordered_map<char, int> mp;
        
        
        for(auto c:s){
            mp[c]++;
        }

        for(int i =0; i<n; i++){
            if(mp[s[i]]<k){
                int left = longestSubstring(s.substr(0,i), k);
                int right = longestSubstring(s.substr(i+1), k);

                return max(left, right);
            }
        }
        return s.size();
        
    }
};

File: 0395-longest-substring-with-at-least-k-repeating-characters/0395-longest-substring-with-at-least-k-repeating-characters.py
class Solution:
    def longestSubstring(self, s: str, k: int) -> int:
        n = len(s)
        d = dict()

        for c in s:
            if c not in d:
                d[c] = 1
            else:
                d[c]+=1

        for i in range(n):
            if d[s[i]] < k:

                left = self.longestSubstring(s[:i], k)
                right = self.longestSubstring(s[i+1:], k)

                return max(left, right)

        return len(s)
        